temporal-sdk/src/main/java/io/temporal/client/schedules/ScheduleIntervalSpec.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (C) 2022 Temporal Technologies, Inc. All Rights Reserved.
+ *
+ * Copyright (C) 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this material except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.temporal.client.schedules;
+
+import com.google.common.base.Preconditions;
+import java.time.Duration;
+import java.util.Objects;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+/**
+ * Specification for scheduling on an interval. Matching times are expressed as
+ *
+ * <p>epoch + (n * every) + offset.
+ */
+public final class ScheduleIntervalSpec {
```

Comment:
Why does `ScheduleCalendarSpec` require a builder but `ScheduleIntervalSpec` I can just use directly?

Reply:
Originally I had `ScheduleIntervalSpec ` use a builder but it provided no value because there is one required parameter and one optional, just made the code more verbose.

Reply:
I think we should be consistent. I'd be ok with no builder for this either or a builder for both. It is confusing that once you hit a field count you have to create objects a completely different way.

Reply:
I think having a high optional field count is a valid reason  to prefer a builder over overloads on a constructor

Reply:
It may be a valid reason all else things not considered, but consistency deserves to be considered. I don't think one type of spec should have to be created one way and another the other way for similar objects.

Reply:
I also hate the builder pattern :), but we need to think of developer experience first. Different objects are different. Just because one object follows a pattern because it is more optimal to express how it is created doesn't mean other objects near it need to follow the same pattern. Having tried all the options suggested already I found the current one gave the best developer experience to actually write schedules .

Reply:
I am fine if we want to add a builder with a required parameter to `ScheduleIntervalSpec`

Reply:
I do think `ScheduleCalendarSpec ` benefits from a builder because a constructor would have tons of parameters with the same type next to each other

Reply:
Question: How are you going to handle a new optional field on `ScheduleIntervalSpec`? You gonna add a constructor for every one?

Maybe you should use a builder everywhere.